src/indices/indexed-class.lisp

```diff
@@ -268,11 +278,25 @@ also index subclasses of the class to which the slot belongs, default is T")
 				:key #'index-holder-name)))
 	  (when old-holder
 	    (index-reinitialize (index-holder-index holder)
-				(index-holder-index old-holder))))))))
+				            (index-holder-index old-holder))))))))
+
 
+#-lispworks
 (defmethod reinitialize-instance :before ((class indexed-class) &key)
   (setf (indexed-class-old-indices class) (indexed-class-indices class)
-	(indexed-class-indices class) nil))
+	    (indexed-class-indices class) nil))
+
+#+lispworks
+(defmethod compute-slots :around ((class indexed-class))
+  ;; Lispworks doesn't call
+  (setf (indexed-class-old-indices class) (indexed-class-indices class)
+	    (indexed-class-indices class) nil)
+  (let ((slots (call-next-method)))
+    (compute-class-indices-with-slots
+     class (indexed-class-index-definitions class)
+     slots)
+    (reinitialize-class-indices class)
+    slots))
```

Comment:
One more change. On LW, I occassionally found complete indexes disappearing. I debugged and it came down to this: I have a class B  that has an indexed slot that extends class A that also has an indexed slot. (e.g. A could just be bknr.datastore:store-object). Now I recompile the expresson for A. Then the indices for B disappear. 

Digging down, it appears on LW, reinitialize-instance on A, does not call reinitialize-instance on B. It does call finalize-inheritence on B though, which calls COMPUTE-SLOS etc. Therefore I modified the code to use COMPUTE-SLOTS to update the index. In theory this might work on other Lisps, but I didn't want to take a chance, so I limited this change to Lispworks.

Reply:
This last change is buggy, it doesn't work correctly when loading pre-compiled FASLs. Looking into it, and will update this again.